crates/primitives/Cargo.toml

```diff
@@ -19,7 +19,7 @@ derive_more = { version = "0.99", default-features = false, features = ["from",
 ink_prelude = { version = "4.0.0-alpha.1", path = "../prelude/", default-features = false }
 scale = { package = "parity-scale-codec", version = "3", default-features = false, features = ["derive", "full"] }
 scale-info = { version = "2", default-features = false, features = ["derive"], optional = true }
-sha2-const = { version = "0.1.2", default-features = false }
+const-fnv1a-hash = "1.1.0"
```

Comment:
Might it even possible to just put it under `build-dependencies` as it is only used in a `const` context?

Reply:
hmm, what happens if I want to use the metadata crate at runtime to create the ABI file?

Reply:
But it is still a `const` function. So it can only be evaluated during compilation. It won't work on values only known at runtime. How are dynamic keys hashes by the way (map keys)? They can't be hashes by this crate obviously. Is there still some SHA in there somewhere?

Reply:
Sure, but if I wanted to use `ink_primitives::KeyComposer::from_str(...)` for example in another project. How would this still be possible if we move all of it into a build script?

Reply:
Point is you can't even do that with the current code as `from_str` is a `const` function. If you want to run that function on a runtime value you would need to add a non-const function and this would be the time when you move the dependency from `build-dependencies` to `dependencies`.

Reply:
I'm not sure whether I understand correctly. `const` function can be run at runtime too?

Reply:
No. The function will always be executed during compilation of the program.